src/Flex.php

```diff
@@ -757,18 +757,42 @@ public function fetchRecipes(array $operations): array
             }
 
             $noRecipe = !isset($manifests[$name]) || (isset($manifests[$name]['not_installable']) && $manifests[$name]['not_installable']);
-            if ($noRecipe && 'symfony-bundle' === $package->getType()) {
-                $manifest = [];
-                $bundle = new SymfonyBundle($this->composer, $package, $job);
-                if (null === $devPackages) {
-                    $devPackages = array_column($this->composer->getLocker()->getLockData()['packages-dev'], 'name');
-                }
-                $envs = \in_array($name, $devPackages) ? ['dev', 'test'] : ['all'];
-                foreach ($bundle->getClassNames() as $class) {
-                    $manifest['manifest']['bundles'][$class] = $envs;
+            if ($noRecipe) {
+                $bundles = [];
+
+                $extra = $package->getExtra();
+                if (isset($extra['symfony']['bundles'])) {
+                    $extraBundles = $extra['symfony']['bundles'];
+                    if (!\is_array($extraBundles)) {
+                        throw new \InvalidArgumentException(sprintf('Entry "bundles" under "extra.symfony" in the composer.json of package "%s" must be an array, "%s" given.', $package->getName(), \gettype($extraBundles)));
+                    }
+                    foreach ($extraBundles as $bundleClass => $envs) {
+                        if (!\is_array($envs)) {
+                            throw new \InvalidArgumentException($bundleClass.' environments should be an array.');
+                        }
+                        foreach ($envs as $env) {
+                            if (!\is_string($env)) {
+                                throw new \InvalidArgumentException('Each environment should be a string.');
```

Comment:
`Bundle "%s" under "extra.symfony.bundles" should list Symfony envs as strings, %s found in the composer.json of package "%s".`